larray/io/session.py

```diff
@@ -67,6 +69,15 @@ def close(self):
         """
         raise NotImplementedError()
 
+    def _create_temporay_file(self):
+        if self.overwrite_file and os.path.isfile(self.fname):
+            self.tmp_file = '{}~{}'.format(*os.path.splitext(self.fname))
+
+    def _delete_temporary_file(self):
```

Comment:
rename to _rename_temporary_file (it is a rename with overwrite -- so more a rename than a delete) and call it from dump_arrays instead?